src/utils/arrow.rs

```diff
@@ -136,12 +139,20 @@ pub(crate) fn arrow_schema_from_column_names_and_range(
     column_names: &[String],
     row_idx: usize,
     row_limit: usize,
+    selected_columns: &SelectedColumns,
 ) -> FastExcelResult<Schema> {
     let mut fields = Vec::with_capacity(column_names.len());
 
-    for (col_idx, name) in column_names.iter().enumerate() {
-        let col_type = get_arrow_column_type(range, row_idx, row_limit, col_idx)?;
-        fields.push(Field::new(&alias_for_name(name, &fields), col_type, true));
+    for (idx, name) in column_names.iter().enumerate() {
+        // If we have an index for the given column, extract it and add it to the schema. Otherwise,
+        // just ignore it
+        if let Some(col_idx) = match selected_columns {
```

Comment:
I reckon the code is not optimal performance-wise here but we probably don't care to build the schema

Reply:
Yeah I was wondering if I should remove the `TryFrom<&ExcelSheet>` implem for `RecordBatch` to avoid re-instantiating `column_names` as well, but this isn't really a hot path

Reply:
Yeah it troubles me a bit but we can go on with this for now and rethink this in the future.